cirq/circuits/circuit.py

```diff
@@ -1861,15 +1874,24 @@ def _draw_moment_in_diagram(
         if x > max_x:
             max_x = x
 
-    global_phase = np.product([
-        complex(e.coefficient)
-        for e in moment
-        if isinstance(e, ops.GlobalPhaseOperation)
-    ])
-    if global_phase != 1:
+    global_phase = None
+    tags: List[Any] = []
+    for op in moment:
+        if isinstance(op, ops.TaggedOperation):
+            tags.extend(op.tags)
+            op = op.sub_operation
```

Comment:
Does this collect tags from all ops in the moment?  Add a test where the non-global phase gate has a tag.

Reply:
Oops, yes.  Good catch.

Also, added test that has 2 moments, with non-global gates on each moment.